src/tools/createAwsElasticSearchIndexes.js

```diff
@@ -0,0 +1,203 @@
+    const finishCreatingIndex = moment();
+
+    // Get activity reports.
+    const startGettingReports = moment();
+
+    /* We need to use raw sql here otherwise we timeout and run out of memory. */
+    const reportsSql = `
+    -- Create AR temp table.
+    SELECT
+    ar."id",
+    ar."context",
+    ar."startDate",
+    ar."endDate"
+    INTO TEMP TABLE temp_ars
+    FROM "ActivityReports" ar
+    WHERE ar."calculatedStatus" = 'approved'
+    ORDER BY ar."id" DESC;
+
+    -- Recipient Next Steps.
+    SELECT
+    rns."activityReportId",
+    rns."note"
+    INTO TEMP TABLE temp_recipient_next_steps
+    FROM "NextSteps" rns
+    WHERE rns."noteType" ='RECIPIENT'
+    AND rns."activityReportId" IN (SELECT id FROM temp_ars)
+    GROUP BY rns."activityReportId", rns."note"
+    ORDER BY rns."activityReportId";
+
+    -- Specialist Next Steps.
+    SELECT
+    rns."activityReportId",
+    rns."note"
+    INTO TEMP TABLE temp_specialist_next_steps
+    FROM "NextSteps" rns
+    WHERE rns."noteType" ='SPECIALIST'
+    AND rns."activityReportId" IN (SELECT id FROM temp_ars)
+    GROUP BY rns."activityReportId", rns."note"
+    ORDER BY rns."activityReportId";
+
+    -- Goals.
+    SELECT
+    arg."activityReportId",
+    arg."name"
+    INTO TEMP TABLE temp_goals
+    FROM "ActivityReportGoals" arg
+    WHERE arg."activityReportId" IN (SELECT id FROM temp_ars)
+    GROUP BY arg."activityReportId", arg."name"
+    ORDER BY arg."activityReportId";
+
+    -- Objectives.
+    SELECT
+    aro."activityReportId",
+    aro."title",
+    aro."ttaProvided"
+    INTO TEMP TABLE temp_objectives
+    FROM "ActivityReportObjectives" aro
+    WHERE aro."activityReportId" IN (SELECT id FROM temp_ars)
+    GROUP BY aro."activityReportId", aro."title", aro."ttaProvided"
+    ORDER BY aro."activityReportId";`;
+
+    // Query DB.
+    let reportsToIndex;
+    let recipientNextStepsToIndex;
+    let specialistNextStepsToIndex;
+    let goalsToIndex;
+    let objectivesToIndex;
+    await sequelize.transaction(async (transaction) => {
+      // Create Temp Tables.
+      await sequelize.query(
+        reportsSql,
+        { transaction },
+      );
+
+      // Reports.
+      reportsToIndex = await sequelize.query(
+        'SELECT * FROM temp_ars;',
+        { transaction },
+      );
+
+      // Recipient Steps.
+      recipientNextStepsToIndex = await sequelize.query(
+        'SELECT * FROM temp_recipient_next_steps;',
+        { transaction },
+      );
+
+      // Specialist Steps.
+      specialistNextStepsToIndex = await sequelize.query(
+        'SELECT * FROM temp_specialist_next_steps;',
+        { transaction },
+      );
+
+      // Goals.
+      goalsToIndex = await sequelize.query(
+        'SELECT * FROM temp_goals;',
+        { transaction },
+      );
+
+      // objectives.
+      objectivesToIndex = await sequelize.query(
+        'SELECT * FROM temp_objectives;',
+        { transaction },
+      );
+    });
```

Comment:
I don't see how this would have had any memory issues if this were done using the model system as described https://sebhastian.com/sequelize-group-by/

I wrote this in github so there might be some typos, but this should be a bit faster and more in band to the model system.
```suggestion
    // Query DB.
    let reportsToIndex;
    let recipientNextStepsToIndex;
    let specialistNextStepsToIndex;
    let goalsToIndex;
    let objectivesToIndex;
    await sequelize.transaction(async (transaction) => {

      // Reports.
      reportsToIndex = await ActivityReport.findAll({
        attributes: ['id', 'context', 'startDate', 'endDate'],
        where: { calculatedStatus: 'approved'},
        order: ['id', 'ASC'],
        raw: true,
        transaction,
      });

      // Recipient Steps.
      recipientNextStepsToIndex  = await NextStep.findAll({
        attributes: ['activityReportId', 'note'],
        where: {
          noteType: 'RECIPIENT',
          activityReportId: { [Op.in]: reportsToIndex .map((report) => report.id) },
        },
        group: ['activityReportId', 'note'],
        order: ['activityReportId', 'ASC'],
        raw: true,
        transaction,
      });

      // Specialist Steps.
      specialistNextStepsToIndex = await NextStep.findAll({
        attributes: ['activityReportId', 'note'],
        where: {
          noteType: 'SPECIALIST',
          activityReportId: { [Op.in]: reportsToIndex .map((report) => report.id) },
        },
        group: ['activityReportId', 'note'],
        order: ['activityReportId', 'ASC'],
        raw: true,
        transaction,
      });

      // Goals.
      goalsToIndex = await ActivityReportGoal.findAll({
        attributes: ['activityReportId', 'name'],
        where: {
          activityReportId: { [Op.in]: reportsToIndex .map((report) => report.id) },
        }.
        group: ['activityReportId', 'name'],
        order: ['activityReportId', 'ASC'],
        raw: true,
        transaction,
      });

      // objectives.
      objectivesToIndex = await ActivityReportObjective.findAll({
        attributes: ['activityReportId', 'title', 'ttaProvided'],
        where: {
          activityReportId: { [Op.in]: reportsToIndex .map((report) => report.id) },
        }.
        group: ['activityReportId', 'title', 'ttaProvided'],
        order: ['activityReportId', 'ASC'],
        raw: true,
        transaction,
      });
    });
```

Reply:
I'm going with if it ain't broke don't fix it.

My initial attempt was to use the model approach but it timed out jest and ran out of memory.

If you don't see anything major wrong with this I will keep as is.

Reply:
Let me know

Reply:
@GarrettEHill  Did a quick time compare (granted its only 1 pass each)
![image](https://user-images.githubusercontent.com/84350609/208192083-8fe084fd-35ba-4702-88b7-321e8acc15bc.png)